webview-ui/src/AzureServiceOperator/AzureServiceOperator.tsx

```diff
@@ -1,50 +1,43 @@
 import { VSCodeLink } from "@vscode/webview-ui-toolkit/react";
 import styles from "./AzureServiceOperator.module.css";
-import { InitialState, ToWebViewMsgDef } from "../../../src/webview-contract/webviewDefinitions/azureServiceOperator";
-import { getWebviewMessageContext } from "../utilities/vscode";
-import { useEffect, useReducer } from "react";
-import { InstallStepStatus, createState, updateState, userMessageHandler, vscodeMessageHandler } from "./helpers/state";
+import { InitialState } from "../../../src/webview-contract/webviewDefinitions/azureServiceOperator";
+import { useEffect } from "react";
+import { InstallStepStatus, stateUpdater, vscode } from "./helpers/state";
 import { Progress, StepWithDescription } from "./Progress";
-import { getEventHandlers, getMessageHandler } from "../utilities/state";
-import { UserMsgDef } from "./helpers/userCommands";
+import { getStateManagement } from "../utilities/state";
 import { Inputs } from "./Inputs";
 import { getRequiredInputs } from "./helpers/inputs";
 
-export function AzureServiceOperator(props: InitialState) {
-    const vscode = getWebviewMessageContext<"aso">();
-
-    const [state, dispatch] = useReducer(updateState, createState());
-
-    const userMessageEventHandlers = getEventHandlers<UserMsgDef>(dispatch, userMessageHandler);
+export function AzureServiceOperator(initialState: InitialState) {
```

Comment:
Note: this tries to establish a convention of explicitly naming the props as `initialState` so that it's clear it should only be used to set up state management (`getStateManagement(stateUpdater, initialState);`).

It is not intended to be referred to later in the component (`state` should be used instead).